extensions/json-jackson/src/main/java/io/deephaven/json/jackson/ObjectMixin.java

```diff
@@ -0,0 +1,421 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json.jackson;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.json.ObjectFieldOptions;
+import io.deephaven.json.ObjectFieldOptions.RepeatedBehavior;
+import io.deephaven.json.ObjectOptions;
+import io.deephaven.qst.type.Type;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+
+final class ObjectMixin extends Mixin<ObjectOptions> {
+
+    public ObjectMixin(ObjectOptions options, JsonFactory factory) {
+        super(factory, options);
+    }
+
+    @Override
+    public Stream<Type<?>> outputTypesImpl() {
+        return options.fields()
+                .stream()
+                .map(ObjectFieldOptions::options)
+                .map(this::mixin)
```

Comment:
Why don't we make the recursive mixins at construction time?

Reply:
Done